src/background/messaging/HttpConnection.ts

```diff
@@ -9,31 +9,22 @@ import {
   formatJsonRpcError,
   isJsonRpcRequest,
 } from '@walletconnect/jsonrpc-utils';
-import { networksStore } from 'src/modules/networks/networks-store.background';
-import type { ChainId } from 'src/modules/ethereum/transactions/ChainId';
 
 export class HttpConnection extends EventEmitter {
-  private chainId: ChainId;
+  private url: string;
 
-  constructor({ chainId }: { chainId: ChainId }) {
+  constructor({ url }: { url: string }) {
     super();
-    /** TODO: Should we save just the URL instead of chainId? */
-    this.chainId = chainId;
+    this.url = url;
   }
 
-  async send(
-    request: JsonRpcPayload,
-    _context: unknown
-  ): Promise<JsonRpcResult | JsonRpcError> {
+  async send(request: JsonRpcPayload): Promise<JsonRpcResult | JsonRpcError> {
     if (!isJsonRpcRequest(request)) {
```

Comment:
Looks like this check for `isJsonRpcRequest` is done twice (not really big performance impact though :))
The function returned by the `createHttpConnectionMessageHandler` already checks that message before sending it to `HttpConnection`

Reply:
But we don't know that here